pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
```

Comment:
All slices are passed by pointer, so you'll need to revert those as well.

On Mon, Jun 6, 2016 at 10:39 AM, Gabe Montero <notifications@github.com>
wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/9167#discussion_r65901447>:
>
> > @@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
> >  	if spec.Template == nil {
> >  		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
> >  	} else {
> > +		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
>
> Yeah, the span of mutation had originally occurred to me as well, but
> since the parameter into ValidateDeploymentConfigSpec is NOT a pointer,
> my understanding was that golang would pass in a copy of the param to the
> method.
>
> Looking again though, while the DeploymentConfigSpec passed in is not a
> pointer, the underlying Template field is a pointer. Perhaps then even
> with DeploymentConfigSpec parameter being a "copy", the Template pointer
> field with be the same address in both DeploymentConfigSpec copies ... as
> I type, I'm thinking yes, but confirmation / correction of my train of
> thought here would be great.
>
> I'll start working on the changes in the interim.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9167/files/7108af0c910f3080a151b4fbd222c31265396ccb#r65901447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p6agQgWm-H2gYfbesbJMRIW-2rxbks5qJDE1gaJpZM4It4tM>
> .
>